upload_components/upload.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+IFS=';' read -ra DIRECTORIES <<<"${COMPONENTS_DIRECTORIES:-.}"
+NAMESPACE=${COMPONENTS_NAMESPACE:-espressif}
+
+NUMBER_OF_DIRECTORIES="${#DIRECTORIES[@]}"
+echo "Processing $NUMBER_OF_DIRECTORIES components"
+
+for ITEM in "${DIRECTORIES[@]}"; do
+    FULL_PATH="${GITHUB_WORKSPACE?}/${ITEM}"
+    if [ "$NUMBER_OF_DIRECTORIES" -eq "1" ]; then
+        NAME=${COMPONENT_NAME?}
+    else
+        NAME=$(basename "$(realpath "${FULL_PATH}")")
+    fi
+
+    echo "Processing component \"$NAME\" at $ITEM"
+    python3 -m idf_component_manager upload-component --path="${FULL_PATH}" --namespace="${NAMESPACE}" --name="${NAME}"
```

Comment:
> If the behavior is to reject subsequent uploads with the same version, then this will fail unless the user modifies all component versions in the repository.

Before preparing an archive or uploading a version, the component manager checks if the version mentioned in the manifest is already uploaded. If it's exit code 144 (Nothing to do) is returned. Uploading script handles this return code as a successful result.

So for the case of multiple components in one repo, it checks all components every time but tries to upload only if the version has changed.

> If the behavior is to ignore re-uploaded versions (without producing an error) then the component developer might change the code and forget to change the version, then get confused why the code in the component service hasn't been updated.

We can add another linting action, to check for the hash sum of the component, for the cases forgotten version changes (a ticket is created for it). This action may run not only on the main branch but on all branches. 

It would be handy to make this check a non-fatal one, but unfortunately, it's [not possible yet](https://github.com/actions/toolkit/issues/399)  to make a "pass with notice" workflow with Github Actions. 

>  make changes while the version is still 0.1.0, and finally change the version to 0.2.0 to trigger the upload?

The expected flow is to make a version bump commit to the main branch as soon as the version is ready for upload. But the flow will make the CI workflow on the main branch red if the linter mentioned above is turned on.

Reply:
> Or should they skip the -dev version, make changes while the version is still 0.1.0, and finally change the version to 0.2.0 to trigger the upload?

I would like to keep the current behavior of when a developer should keep the old `0.1.0` version during changes and then make a  version-bump commit with `0.2.0` to upload. @igrr Is this flow ok?

Reply:
It looks okay as long as it is documented  somewhere! At the moment the workflow for the component developer is not clear from upload_components/README.md.

Reply:
I added `skip_pre_release` parameter to allow development flow with pre-release versions as an alternative and added a description to the readme.

Reply:
@igrr  @projectgus  Could you please take one more look and merge, so we can start using this action uploading existing components